### PR TITLE
[Kamino] Core

### PR DESCRIPTION
**Pull request by aserifi**
## DR Legs closed-loop hold-pose and walk task

Depends on https://github.com/newton-physics/newton/pull/3220
 
Simplified many things leaving the pure DR Legs task here.
 
- Adds the Disney DR Legs closed-loop biped (Kamino solver) and a `Isaac-DrLegs-HoldPose-v0` hold-pose task as well as `Isaac-DrLegs-Walk-v0` walking task.
- Uses the `ArticulationView` 
- Fixes Kamino environment resets directly in the kamion_manager
- DR Legs USD is loaded from newton-assets

### CODE REVIEW

**Review by AntoineRichard — APPROVED** [review]

**Comment by AntoineRichard** [comment]
Thanks @aserifi ! Let's wait for 3220 to be merged, then we will likely need to bump the newton commit too.

File: source/isaaclab_newton/isaaclab_newton/physics/kamino_manager.py [hunk 1]
```diff
@@ -14,14 +14,38 @@
...
 from isaaclab.physics import PhysicsManager
-from isaaclab.utils.timer import Timer
 
 from .kamino_manager_cfg import KaminoSolverCfg
 from .newton_manager import NewtonManager
 
 logger = logging.getLogger(__name__)
 
 
+def _model_has_loop_closing_joints(model: Model) -> bool:
```
Comment (rubengrandia) [inline source/isaaclab_newton/isaaclab_newton/physics/kamino_manager.py:24]: I did not find an equivalent helper function in Newton. This could potentially move to upstream Newton at some point, but that would require additional thoughts around if this should be done in warp, on a per-articulation basis, etc. Trying to keep it simple by keeping it locally here for now.

File: source/isaaclab_newton/isaaclab_newton/physics/newton_manager.py [hunk 2]
```diff
@@ -1645,7 +1689,7 @@ def _run_solver_substeps(cls, contacts) -> None:
                     cls._collision_pipeline.collide(cls._state_0, contacts)
         else:
             cfg = PhysicsManager._cfg
-            need_copy_on_last = (cfg is not None and cfg.use_cuda_graph) and cls._num_substeps % 2 == 1  # type: ignore[union-attr]
+            need_copy_on_last = cfg is not None and cls._num_substeps % 2 == 1
```
Comment (rubengrandia) [inline source/isaaclab_newton/isaaclab_newton/physics/newton_manager.py:1708]: Copying the relevant explanation for this from https://github.com/isaac-sim/IsaacLab/pull/5984

**What changed:**

- **Bug fix — `use_cuda_graph` gating the double-buffer swap.** In
  `_run_solver_substeps`, `need_copy_on_last` was gated on `cfg.use_cuda_graph`.
  With double-buffered state (Kamino uses `_use_single_state=False`) and an odd
  substep count, this left the canonical state in `_state_1` while reset writes
  (which target the `_state_0`-bound data layer) landed in the stale buffer when
  CUDA graphs were disabled. The gate is removed so the copy-on-last always runs
  for odd substep counts.

File: source/isaaclab_assets/isaaclab_assets/robots/dr_legs.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    copy_from_source=False,
+)
+
+
+_DR_LEGS_INIT_STATE = ArticulationCfg.InitialStateCfg(
+    pos=(0.0, 0.0, 0.27),
+    # Closed-loop FK is only valid at the assembled reference (all joint coords zero).
+    joint_pos={".*": 0.0},
+    joint_vel={".*": 0.0},
+)
```
Comment (AntoineRichard) [inline source/isaaclab_assets/isaaclab_assets/robots/dr_legs.py]: Assets typically inline these in the ArticulationCfg.

**Review by AntoineRichard — APPROVED** [review]
LGTM. Thanks for all the work!

File: source/isaaclab_newton/isaaclab_newton/physics/kamino_manager.py (hunk 1 above)
Comment (AntoineRichard) [reply source/isaaclab_newton/isaaclab_newton/physics/kamino_manager.py:24]: I agree, @camevor for suggestions?

File: source/isaaclab_tasks/isaaclab_tasks/contrib/dr_legs/mdp/__init__.py
Comment (ooctipus) [inline source/isaaclab_tasks/isaaclab_tasks/contrib/dr_legs/mdp/__init__.py:1]: can you use lazy export in __init__.py, and put actual export in __init__.pyi?

File: source/isaaclab_assets/isaaclab_assets/robots/dr_legs.py (hunk 3 above)
Comment (rubengrandia) [reply source/isaaclab_assets/isaaclab_assets/robots/dr_legs.py]: done

File: source/isaaclab_tasks/isaaclab_tasks/contrib/dr_legs/mdp/__init__.py
Comment (rubengrandia) [reply source/isaaclab_tasks/isaaclab_tasks/contrib/dr_legs/mdp/__init__.py:1]: done

File: source/isaaclab_newton/isaaclab_newton/physics/kamino_manager.py (hunk 1 above)
Comment (camevor) [reply source/isaaclab_newton/isaaclab_newton/physics/kamino_manager.py:24]: Since this function simply checks this specific detail based on a publicly documented contract, I don't see any issue with it. We could consider adding a host-side field on Model purely for convenience, but I think keeping this function here is totally fine.

File: source/isaaclab_newton/isaaclab_newton/physics/kamino_manager.py [hunk 4]
```diff
@@ -116,43 +131,29 @@ def step(cls) -> None:
...
-                logger.info("Newton CUDA graph capture deferred until first step() (RTX active)")
+        # Detect closed-loop articulations
+        cls._has_loop_joints = False
+        _art_start = getattr(model, "articulation_start", None)
+        _art_end = getattr(model, "articulation_end", None)
+        if _art_start is not None and _art_end is not None:
+            _art_start_np = _art_start.numpy()
+            _art_end_np = _art_end.numpy()
+            if _art_end_np.shape[0] > 0:
+                cls._has_loop_joints = bool((_art_start_np[1:] > _art_end_np).any())
```
[the remark this answers is not in the document]
Comment (aserifi) [reply source/isaaclab_newton/isaaclab_newton/physics/kamino_manager.py]: thats not the case

File: source/isaaclab_tasks/isaaclab_tasks/contrib/dr_legs/mdp/rewards.py [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    asset_cfg: SceneEntityCfg = SceneEntityCfg("robot"),
+    std: float = 0.5,
+) -> torch.Tensor:
+    asset = env.scene[asset_cfg.name]
+    joint_ids = asset_cfg.joint_ids
+    error = asset.data.joint_pos[:, joint_ids] - asset.data.default_joint_pos[:, joint_ids]
+    return torch.exp(-torch.sum(torch.square(error), dim=1) / std**2)
+
+
+def base_lin_vel_l2(env: ManagerBasedRLEnv, asset_cfg: SceneEntityCfg = SceneEntityCfg("robot")) -> torch.Tensor:
```
Comment (ooctipus) [inline source/isaaclab_tasks/isaaclab_tasks/contrib/dr_legs/mdp/rewards.py:88]: looks like some of these mdp has exact equation in core mdp, a suggestion from me is to reuse them, but I am also fine with define it here with dr_legs

Comment (aserifi) [reply source/isaaclab_tasks/isaaclab_tasks/contrib/dr_legs/mdp/rewards.py:88]: I didn't find exact equivalents. Every term is slightly different. E.g., the closest to `base_lin_vel_l2` is `lin_vel_z_l2`, which only considers the z-axis. We will do more reward tuning and testing in later PRs, where I can try to better align them with those in core.
